src/Transactions/AccountTransactionPayload.cs

```diff
@@ -0,0 +1,51 @@
+using Concordium.Sdk.Types;
+
+namespace Concordium.Sdk.Transactions;
+
+/// <summary>
+/// Represents the payload of account transaction.
+///
+/// Inheriting records should implement data specific to the transaction they
+/// model as well as helpers for constructing serialized transaction payloads
+/// to be sent to the Concordium node.
+/// </summary>
+public abstract record AccountTransactionPayload<T>
+    where T : AccountTransactionPayload<T>
```

Comment:
I’m not sure I understand why this is added. It will always be true.

I tried remove this and made the code compile. IMHO I can’t see why this is getting us more type safety. All places used we just use we expect a `AccountTransactionPayload`.